### DISCUSSION TITLE
Can I Download Vercel Postgres Database?

### DISCUSSION THREAD

**Discussion by StokleyO** [vercel/community#5090] [Help] [answered]
### Question

I currently host my PostgreSQL Database for my project through Vercel. Is there any way that I can download a copy of this database? I want to have access to all of the information as a just-in-case measure

**Comment by amyegan** [comment, marked as answer]
Hey @StokleyO. You can use `pg_dump` to [download a copy of your database](https://vercel.com/docs/storage/vercel-postgres/faq#external-data-migration). The postgres connection [environment variables](https://vercel.com/storage/vercel-postgres/quickstart#review-what-was-created) created in your Vercel project can be used to define the connection string.

https://neon.tech/docs/import/import-from-postgres#export-data-with-pgdump

**Reply by StokleyO** [reply]
Thank you so much @amyegan

**Reply by StokleyO** [reply]
@amyegan I just want to clarify also, if I run pg_dump, will my current database in Vercel be affected in any way? I want it to still be up & running like regular, I would just like a local copy

**Reply by amyegan** [reply]
Good thing to check! It's a backup utility, so it doesn't affect the data in your active database. You can find more info at [postgresql.org/docs/current/app-pgdump.html](https://www.postgresql.org/docs/current/app-pgdump.html)

> pg_dump is a utility for backing up a PostgreSQL database. It makes consistent backups even if the database is being used concurrently. pg_dump does not block other users accessing the database (readers or writers).

https://stackoverflow.com/questions/50125414/postgres-pg-dump-effect-on-production-db

**Reply by StokleyO** [reply]
@amyegan thank you for your help again Amy. I'm trying to run pg_dump as a query and am getting the following error. Is this not the correct way to use it? how else could I run the command? very sorry if the answer is obvious
<img width="992" alt="Screenshot 2024-01-02 at 2 15 29 PM" src="https://github.com/vercel/community/assets/83097972/b8469dfc-1a67-4591-8713-1d1dc873baf0">

**Reply by StylianosS** [reply]
Any solution?

**Reply by amyegan** [reply]
You should use `pg_dump` in the location where you want to save the backup. In my case, this was on my local computer. I used the database related environment variables from Vercel project to define the connection string.

`pg_dump -Fc -v -d <source_database_connection_string> -f <dump_file_name>`

https://neon.tech/docs/import/import-from-postgres#export-data-with-pgdump

**Reply by StylianosS** [reply]
So i open terminal on my windows?

**Reply by amyegan** [reply]
exactly 🙂

**Reply by StylianosS** [reply]
pg_dump is an unnkown command... All the stuff about the pg_dump on web is outdated so if you have some guide on how to do it; i will appreciate it.

**Reply by StylianosS** [reply]
Solved! I installed postgres on my windows and then with some tutorials on youtube i learned how to use pg_dump and everything worked correctly. Thank you!

[a comment by an automated account is not shown]
